src/Compilers/Core/Portable/Emit/CommonPEModuleBuilder.cs

```diff
@@ -197,6 +197,8 @@ public IEnumerable<Cci.INamespaceTypeDefinition> GetTopLevelTypeDefinitionsCore(
         /// </summary>
         public abstract MultiDictionary<Cci.DebugSourceDocument, Cci.DefinitionWithLocation> GetSymbolToLocationMap();
 
+        public abstract List<(Cci.IDefinition definition, List<Cci.DebugSourceDocument> document)> GetTypeDocument(EmitContext context);
```

Comment:
>GetTypeDocument [](http://example.com/codeflow?start=99&length=15)

The name could be better: it returns a list so it should be plurar and it could be a bit more specific (type is pretty general as types can be references or definitions).

E.g. GetTypeDefinitionDocuments